compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs

```diff
@@ -144,45 +205,61 @@ constr2rst RecordC{..} = T.unlines
 >      - `Text`
 >      - and text
 -}
-fieldTable :: [FieldDoc] -> T.Text
-fieldTable []  = ""
-fieldTable fds = T.unlines $ -- NB final empty line is essential and intended
-  [ ".. list-table::", "   :widths: 15 10 30", "   :header-rows: 1", ""]
-  <> map (indent 3) (headerRow <> fieldRows)
+fieldTable :: [FieldDoc] -> RenderOut
+fieldTable []  = mempty
+fieldTable fds = mconcat -- NB final empty line is essential and intended
+    [ renderLines
+        [ ".. list-table::"
+        , "   :widths: 15 10 30"
+        , "   :header-rows: 1"
+        , ""
+        , "   * - Field"
+        , "     - Type"
+        , "     - Description" ]
+    , fieldRows
+    ]
   where
-    headerRow = [ "* - Field"
-                , "  - Type"
-                , "  - Description" ]
-    fieldRows = concat
-       [ [ prefix "* - " $ escapeTr_ (unFieldname fd_name)
-         , prefix "  - " $ type2rst fd_type
-         , prefix "  - " $ maybe " " (docTextToRst . DocText . T.unwords . T.lines . unDocText) fd_descr ] -- FIXME: this makes no sense
+    fieldRows = renderLinesDep $ \env -> concat
+       [ [ prefix "   * - " $ escapeTr_ (unFieldname fd_name)
+         , prefix "     - " $ type2rst env fd_type
+         , prefix "     - " $ maybe " " (docTextToRst . DocText . T.unwords . T.lines . unDocText) fd_descr ] -- FIXME: indent properly instead of this
        | FieldDoc{..} <- fds ]
 
 -- | Render a type. Nested type applications are put in parentheses.
-type2rst :: Type -> T.Text
-type2rst = f (0 :: Int)
+type2rst :: RenderEnv -> Type -> T.Text
+type2rst env = f 0
   where
     -- 0 = no brackets
     -- 1 = brackets around function
     -- 2 = brackets around function AND application
-    f _ (TypeApp _ n []) = unTypename n
-    f i (TypeApp _ n as) = (if i >= 2 then inParens else id) $ T.unwords (unTypename n : map (f 2) as)
-    f i (TypeFun ts) = (if i >= 1 then inParens else id) $ T.intercalate " -> " $ map (f 1) ts
+    f :: Int -> Type -> T.Text
+    f _ (TypeApp a n []) = link a n
+    f i (TypeApp a n as) = (if i >= 2 then inParens else id) $
+        T.unwords (link a n : map (f 2) as)
+    f i (TypeFun ts) = (if i >= 1 then inParens else id) $
+        T.intercalate " -> " $ map (f 1) ts
     f _ (TypeList t1) = "[" <> f 0 t1 <> "]"
     f _ (TypeTuple ts) = "(" <> T.intercalate ", " (map (f 0) ts) <>  ")"
 
+    link :: Maybe Anchor -> Typename -> T.Text
+    link Nothing n = unTypename n
+    link (Just anchor) n =
+        if renderAnchorAvailable env anchor
```

Comment:
Do we do this to guard against bugs in our implementation or because there are valid cases where the anchor does not exists? I suspect the latter but I’m not entirely clear on when that happens so it would be good to document this in a comment (e.g. in the haddocks for `renderAnchorAvailable`).

Reply:
Will do. Anchors may be missing because they are hidden (through the use of the `-- | HIDE` directive), or because they are imported from a different package (no way to link docs across packages).